pkg/apis/kops/validation/validation.go

```diff
@@ -1735,6 +1739,26 @@ func validateContainerdConfig(spec *kops.ClusterSpec, config *kops.ContainerdCon
 	return allErrs
 }
 
+func validateNriConfig(containerd *kops.ContainerdConfig, fldPath *field.Path) (allErrs field.ErrorList) {
+	if !fi.ValueOf(containerd.NRI.Enabled) {
+		return allErrs
+	}
```

Comment:
This is already checked previously.
```suggestion
```

Reply:
Any thoughts about ^^^ ?

Reply:
I have remove it and I'm +1 but now my tests are failing and I'm trying to understand the reason
```

=== RUN   Test_Validate_NriConfig
    validation_test.go:1587: expected error "Forbidden::nri" from { [] {  } {<nil> <nil> <nil> <nil> <nil> <nil> <nil>} <nil>    <nil>  [] [] <nil> <nil> map[] map[] [] [] <nil> 0xc0008cc460 <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> {  [] [] <nil> <nil> <nil>    <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil>} {<nil> <nil>  [] []} <nil> <nil> <nil> map[] [] <nil> <nil> <nil> <nil> <nil> [] <nil> <nil> <nil> <nil> <nil>}, was not found in ["Forbidden::containerd"]
    validation_test.go:1587: unexpected errors from {"" [] {"" "" ""} {%!q(*kops.AWSSpec=<nil>) %!q(*kops.AzureSpec=<nil>) %!q(*kops.DOSpec=<nil>) %!q(*kops.GCESpec=<nil>) %!q(*kops.HetznerSpec=<nil>) %!q(*kops.OpenstackSpec=<nil>) %!q(*kops.ScalewaySpec=<nil>)} %!q(*kops.GossipConfig=<nil>) "" "" "" %!q(*kops.DNSControllerGossipConfig=<nil>) "" [] [] %!q(*string=<nil>) %!q(*string=<nil>) map[] map[] [] [] %!q(*kops.DockerConfig=<nil>) %!q(*kops.ContainerdConfig=&{<nil> <nil> <nil> <nil> map[] <nil> false <nil> 0xc000797b20 <nil> <nil> false 0xc00012ef48}) %!q(*kops.KubeDNSConfig=<nil>) %!q(*kops.KubeAPIServerConfig=<nil>) %!q(*kops.KubeControllerManagerConfig=<nil>) %!q(*kops.CloudControllerManagerConfig=<nil>) %!q(*kops.KubeSchedulerConfig=<nil>) %!q(*kops.KubeProxyConfig=<nil>) %!q(*kops.KubeletConfigSpec=<nil>) %!q(*kops.KubeletConfigSpec=<nil>) %!q(*kops.CloudConfiguration=<nil>) %!q(*kops.ExternalDNSConfig=<nil>) %!q(*kops.NTPConfig=<nil>) %!q(*kops.NodeProblemDetectorConfig=<nil>) %!q(*kops.MetricsServerConfig=<nil>) %!q(*kops.CertManagerConfig=<nil>) {"" "" [] [] %!q(*bool=<nil>) %!q(*kops.TopologySpec=<nil>) %!q(*kops.EgressProxySpec=<nil>) "" "" "" %!q(*bool=<nil>) %!q(*kops.ClassicNetworkingSpec=<nil>) %!q(*kops.KubenetNetworkingSpec=<nil>) %!q(*kops.ExternalNetworkingSpec=<nil>) %!q(*kops.CNINetworkingSpec=<nil>) %!q(*kops.KopeioNetworkingSpec=<nil>) %!q(*kops.WeaveNetworkingSpec=<nil>) %!q(*kops.FlannelNetworkingSpec=<nil>) %!q(*kops.CalicoNetworkingSpec=<nil>) %!q(*kops.CanalNetworkingSpec=<nil>) %!q(*kops.KuberouterNetworkingSpec=<nil>) %!q(*kops.RomanaNetworkingSpec=<nil>) %!q(*kops.AmazonVPCNetworkingSpec=<nil>) %!q(*kops.CiliumNetworkingSpec=<nil>) %!q(*kops.LyftVPCNetworkingSpec=<nil>) %!q(*kops.GCPNetworkingSpec=<nil>)} {%!q(*kops.DNSAccessSpec=<nil>) %!q(*kops.LoadBalancerAccessSpec=<nil>) "" [] []} %!q(*kops.AuthenticationSpec=<nil>) %!q(*kops.AuthorizationSpec=<nil>) %!q(*kops.NodeAuthorizationSpec=<nil>) map[] [] %!q(*kops.AssetsSpec=<nil>) %!q(*kops.IAMSpec=<nil>) %!q(*bool=<nil>) %!q(*kops.TargetSpec=<nil>) %!q(*bool=<nil>) [] %!q(*kops.RollingUpdate=<nil>) %!q(*kops.ClusterAutoscalerConfig=<nil>) %!q(*kops.ServiceAccountIssuerDiscoveryConfig=<nil>) %!q(*kops.SnapshotControllerConfig=<nil>) %!q(*kops.KarpenterConfig=<nil>)}: [containerd: Forbidden: NRI is available starting from version 1.7.0 and above]
    validation_test.go:1587: unexpected errors from {"" [] {"" "" ""} {%!q(*kops.AWSSpec=<nil>) %!q(*kops.AzureSpec=<nil>) %!q(*kops.DOSpec=<nil>) %!q(*kops.GCESpec=<nil>) %!q(*kops.HetznerSpec=<nil>) %!q(*kops.OpenstackSpec=<nil>) %!q(*kops.ScalewaySpec=<nil>)} %!q(*kops.GossipConfig=<nil>) "" "" "" %!q(*kops.DNSControllerGossipConfig=<nil>) "" [] [] %!q(*string=<nil>) %!q(*string=<nil>) map[] map[] [] [] %!q(*kops.DockerConfig=<nil>) %!q(*kops.ContainerdConfig=&{<nil> <nil> <nil> <nil> map[] <nil> false <nil> 0xc000797b20 <nil> <nil> false 0xc00012ef78}) %!q(*kops.KubeDNSConfig=<nil>) %!q(*kops.KubeAPIServerConfig=<nil>) %!q(*kops.KubeControllerManagerConfig=<nil>) %!q(*kops.CloudControllerManagerConfig=<nil>) %!q(*kops.KubeSchedulerConfig=<nil>) %!q(*kops.KubeProxyConfig=<nil>) %!q(*kops.KubeletConfigSpec=<nil>) %!q(*kops.KubeletConfigSpec=<nil>) %!q(*kops.CloudConfiguration=<nil>) %!q(*kops.ExternalDNSConfig=<nil>) %!q(*kops.NTPConfig=<nil>) %!q(*kops.NodeProblemDetectorConfig=<nil>) %!q(*kops.MetricsServerConfig=<nil>) %!q(*kops.CertManagerConfig=<nil>) {"" "" [] [] %!q(*bool=<nil>) %!q(*kops.TopologySpec=<nil>) %!q(*kops.EgressProxySpec=<nil>) "" "" "" %!q(*bool=<nil>) %!q(*kops.ClassicNetworkingSpec=<nil>) %!q(*kops.KubenetNetworkingSpec=<nil>) %!q(*kops.ExternalNetworkingSpec=<nil>) %!q(*kops.CNINetworkingSpec=<nil>) %!q(*kops.KopeioNetworkingSpec=<nil>) %!q(*kops.WeaveNetworkingSpec=<nil>) %!q(*kops.FlannelNetworkingSpec=<nil>) %!q(*kops.CalicoNetworkingSpec=<nil>) %!q(*kops.CanalNetworkingSpec=<nil>) %!q(*kops.KuberouterNetworkingSpec=<nil>) %!q(*kops.RomanaNetworkingSpec=<nil>) %!q(*kops.AmazonVPCNetworkingSpec=<nil>) %!q(*kops.CiliumNetworkingSpec=<nil>) %!q(*kops.LyftVPCNetworkingSpec=<nil>) %!q(*kops.GCPNetworkingSpec=<nil>)} {%!q(*kops.DNSAccessSpec=<nil>) %!q(*kops.LoadBalancerAccessSpec=<nil>) "" [] []} %!q(*kops.AuthenticationSpec=<nil>) %!q(*kops.AuthorizationSpec=<nil>) %!q(*kops.NodeAuthorizationSpec=<nil>) map[] [] %!q(*kops.AssetsSpec=<nil>) %!q(*kops.IAMSpec=<nil>) %!q(*bool=<nil>) %!q(*kops.TargetSpec=<nil>) %!q(*bool=<nil>) [] %!q(*kops.RollingUpdate=<nil>) %!q(*kops.ClusterAutoscalerConfig=<nil>) %!q(*kops.ServiceAccountIssuerDiscoveryConfig=<nil>) %!q(*kops.SnapshotControllerConfig=<nil>) %!q(*kops.KarpenterConfig=<nil>)}: [containerd: Forbidden: NRI is available starting from version 1.7.0 and above]
--- FAIL: Test_Validate_NriConfig (0.00s)
FAIL
FAIL	k8s.io/kops/pkg/apis/kops/validation	0.020s
FAIL
```

Reply:
Better now?

Reply:
Unfortunately no. Seems I had path set incorrectly in my tests and trying to dig out. I will ping you once I have fixed it:)
Thanks for the quick responses.

Reply:
@hakman I fixed the issue now. PTAL when you can.